Valeriia/Movies_database_project/src/App.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { Component } from "react";
+import Header from "./components/Header";
+import Footer from "./components/Footer";
+import Movies from "./components/Movies";
+import SortPannel from "./components/SortPannel";
+import Loader from "./components/Loader";
+import movies from "./movies.json";
+import NotFound from "./components/NotFound";
+import Movie from "./components/Movie";
+import { IMovie } from "./types";
+import "./App.scss";
+
+interface IAppState {
+  movies: IMovie[] | [];
+  isLoading: boolean;
+  tempListOfMovies: IMovie[] | [];
+  filterBy: string;
+  sortBy: string;
+  searchTerm: string;
+  movie?: IMovie;
+}
+
+class App extends Component<{}, IAppState> {
+  state = {
+    movies: [],
+    isLoading: false,
+    tempListOfMovies: [],
+    filterBy: "title",
+    sortBy: "date",
+    searchTerm: "",
+    movie: undefined,
+  };
+
+  onSearchHandler = () => {
+    this.setState({ movie: undefined, isLoading: true, searchTerm: "" });
+    const { searchTerm, filterBy, movies } = this.state;
+    const tempListOfMovies = movies.filter(
+      (movie) => movie[filterBy] === searchTerm
+    );
+    setTimeout(() => {
+      this.setState({ tempListOfMovies, isLoading: false });
+    }, 1000);
+  };
+
+  componentDidMount = () => {
+    this.setState({ isLoading: true });
+
+    setTimeout(() => {
+      this.setState({
+        movies: movies,
+        isLoading: false,
+        tempListOfMovies: movies.sort((a, b) => b.date - a.date),
+      });
+    }, 1000);
+  };
+
+  onChangeFilterByHandler = (filterBy: string) => {
+    this.setState({ filterBy });
+  };
+
+  onChangeHandler = (searchTerm: string) => {
+    this.setState({ searchTerm });
+  };
+
+  onClickSortByHandler = (sortByType: string) => {
+    this.setState({ sortBy: sortByType });
+    const { tempListOfMovies } = this.state;
+    const tempListSortMovies = tempListOfMovies.sort((a, b) => {
+      return b[sortByType] - a[sortByType];
+    });
+    this.setState({ tempListOfMovies: tempListSortMovies });
```

Comment:
if you have to update several fields, you should do it in one call of this.setState. It's bad practice to call setState more than once in methods
```
    this.setState({ 
       sortBy: sortByType 
       tempListOfMovies: tempListSortMovies 
    });
```